src/tests/Microsoft.PowerFx.Interpreter.Tests/RecalcEngineTests.cs

```diff
@@ -732,6 +731,50 @@ public void OptionSetChecks()
             Assert.True(checkResult.IsSuccess);
         }
 
+        [Theory]
+        [InlineData("OptionSetInfo(OptionSet.option_1)", "option_1")]
+        [InlineData("OptionSetInfo(OptionSet.Option1)", "option_1")]
+        [InlineData("OptionSetInfo(Option1)", "option_1")]
+        [InlineData("OptionSetInfo(If(1<0, Option1))", null)]
+        public async void OptionSetInfoTests(string expression, string expected)
+        {
+            var optionSet = new OptionSet("OptionSet", DisplayNameUtility.MakeUnique(new Dictionary<string, string>()
+            {
+                    { "option_1", "Option1" },
+                    { "option_2", "Option2" }
+            }));
+
+            optionSet.TryGetValue(new DName("option_1"), out var option1);
+
+            var symbol = new SymbolTable();
+            var option1Solt = symbol.AddVariable("Option1", FormulaType.OptionSetValue);
+            var symValues = new SymbolValues(symbol);
+            symValues.Set(option1Solt, option1);
+
+            var config = new PowerFxConfig() { SymbolTable = symbol };
+            config.AddOptionSet(optionSet);
+            var recalcEngine = new RecalcEngine(config);
+            
+            var result = await recalcEngine.EvalAsync(expression, CancellationToken.None, symValues);
+            Assert.Equal(expected, result.ToObject());
+        }
+
+        [Fact]
+        public async Task OptionSetInfoNegativeTest()

```

Comment:
>OptionSetInfoNegativeTest [](http://example.com/codeflow?start=26&length=25)

Would also be good to add some negative cases for what we *can't* pass to OptionSetInfo.
Ie, no strings, no enums, no numbers. 
OptionSetInfo(Color.Red) should fail.

Reply:
updated